core/src/test/java/org/elasticsearch/indices/cluster/IndicesClusterStateServiceRandomUpdatesTests.java

```diff
@@ -267,7 +318,7 @@ private IndicesClusterStateService createIndicesClusterStateService() {
         final ThreadPool threadPool = mock(ThreadPool.class);
         final Executor executor = mock(Executor.class);
         when(threadPool.generic()).thenReturn(executor);
-        final MockIndicesService indicesService = new MockIndicesService();
+        final MockIndicesService indicesService = new RecordingIndicesService();
```

Comment:
or just leave it up to the caller to pass the `AllocatedIndices<? extends Shard, ? extends AllocatedIndex<? extends Shard>> indicesService` in.